### PR TITLE
feat: Provide option to send test email to test out email configurati…

### LINKED ISSUE #8991: [Feature]:Option to send test email from admin settings page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

 - Provide an option to send test email from admin settings page.
 - Admin settings should be shown/hidden based on isConfigurable property.

### Why should this be worked on?

Admin can test out the smtp configuration with this option.

### PR DESCRIPTION
**Pull request by sbalaji1192**
…on from admin settings page
## Description
Adds a button to the admin settings ui under 'email' category to test out the email configuration.

Fixes #8991 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/option-to-send-test-mail 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.79 **(0)** | 36.62 **(-0.01)** | 33.52 **(-0.03)** | 55.32 **(0)**
 :red_circle: | app/client/src/api/UserApi.tsx | 65.22 **(0.07)** | 0 **(0)** | 4.55 **(-0.21)** | 64.71 **(0.09)**
 :red_circle: | app/client/src/constants/messages.ts | 74.07 **(-0.09)** | 100 **(0)** | 23.29 **(-0.19)** | 79.62 **(-0.02)**
 :red_circle: | app/client/src/pages/common/ProfileDropdown.tsx | 67.65 **(0)** | 27.27 **(-8.02)** | 0 **(0)** | 67.65 **(0)**
 :sparkles: :new: | **app/client/src/sagas/SuperUserSagas.tsx** | **26.98** | **3.85** | **12.5** | **30.91**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.26 **(0.22)** | 41.25 **(0.83)** | 34.48 **(0)** | 56.22 **(0.26)**
 :x: | ~~app/client/src/sagas/SuperUserSagas.ts~~ | ~~26.53~~ | ~~6.25~~ | ~~16.67~~ | ~~30.23~~</details>

### CODE REVIEW

File: app/client/src/pages/Settings/WithSuperUserHoc.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default function WithSuperUserHOC(
   return function Wrapped(props: RouteComponentProps) {
     const user = useSelector(getCurrentUser);
 
-    if (!user?.isSuperUser) {
+    if (!user?.isSuperUser || !user?.isConfigurable) {
```
Comment (rishabhsaxena) [inline app/client/src/pages/Settings/WithSuperUserHoc.tsx:13]: Hey QQ: wondering what `isConfigurable` signifies?

File: app/client/src/sagas/SuperUserSagas.tsx [hunk 2]
```diff
@@ -91,6 +100,40 @@ function* RestartServerPoll() {
...
+        <>
+          <br />
+          <span onClick={() => window.open(EMAIL_SETUP_DOC, "blank")}>
+            {createMessage(TEST_EMAIL_SUCCESS_TROUBLESHOOT)}
+          </span>
+        </>
+      );
+    }
+    Toaster.show({
+      actionElement,
```
Comment (rishabhsaxena) [inline app/client/src/sagas/SuperUserSagas.tsx:119]: QQ: We're showing the `troubleshoot` toast on success as well, just checking if this is by design?

**Review by rishabhsaxena — DISMISSED** [review]
LGTM

File: app/client/src/pages/Settings/WithSuperUserHoc.tsx (hunk 1 above)
Comment (sbalaji1192) [reply app/client/src/pages/Settings/WithSuperUserHoc.tsx:13]: isConfigurable tells whether the Appsmith env variables can be configured on runtime on the given instance.

File: app/client/src/sagas/SuperUserSagas.tsx (hunk 2 above)
Comment (sbalaji1192) [reply app/client/src/sagas/SuperUserSagas.tsx:119]: Yeah. Two things going on when user wants to send test mail. Browser submits a request to backend for which backend acknowledges right away after scheduling a request to SMTP host. But SMTP host may reject the request for various reasons , so we're showing troubleshoot button. the success toaster just indicates that BE has scheduled a request to SMTP host

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**